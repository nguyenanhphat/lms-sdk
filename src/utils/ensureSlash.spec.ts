describe('Exported functions', () => {
  test('Exported default function', () => {
    const { default: ensureSlash } = require('./ensureSlash');
    expect(ensureSlash).toBeInstanceOf(Function);
  });

  test('Exported "ensureSlash" function', () => {
    const { ensureSlash } = require('./ensureSlash');
    expect(ensureSlash).toBeInstanceOf(Function);
  });
});

describe('Ensure having slash', () => {
  const SAMPLES = [
    ['#/', '#/'],
    ['/', '/'],
    ['//', '//'],
    ['', '/'],
  ];

  const { default: ensureSlash } = require('./ensureSlash');
  test.each(SAMPLES)(
    'Pass %j should be return %j',
    (path: string, expected: string) => {
      expect(ensureSlash(path, true)).toEqual(expected);
    },
  );
});

describe('Ensure NOT having slash', () => {
  const SAMPLES = [
    ['#', '#'],
    ['/', ''],
    ['//', '/'],
    ['', ''],
  ];

  const { default: ensureSlash } = require('./ensureSlash');
  test.each(SAMPLES)(
    'Pass %j should be return %j',
    (path: string, expected: string) => {
      expect(ensureSlash(path, false)).toEqual(expected);
    },
  );
});

describe('Failed cases', () => {
  const SAMPLES = [null, undefined, {}, new Map(), new Set()];

  const { default: ensureSlash } = require('./ensureSlash');
  test.each(SAMPLES)('Pass %j should throw error', (path: any) => {
    function run() {
      ensureSlash(path, true);
    }
    expect(run).toThrowErrorMatchingSnapshot();
  });
});
