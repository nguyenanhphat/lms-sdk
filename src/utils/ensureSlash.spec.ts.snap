// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Failed cases Pass {} should throw error 1`] = `"Path received type is object not is a string"`;

exports[`Failed cases Pass {} should throw error 2`] = `"Path received type is object not is a string"`;

exports[`Failed cases Pass {} should throw error 3`] = `"Path received type is object not is a string"`;

exports[`Failed cases Pass null should throw error 1`] = `"Path received type is object not is a string"`;

exports[`Failed cases Pass undefined should throw error 1`] = `"Path received type is undefined not is a string"`;
